compiler/rustc_lint/src/context.rs

```diff
@@ -1316,6 +1316,40 @@ impl<'tcx> LateContext<'tcx> {
             })
     }
 
+    /// If the given expression is a local binding, find the initializer expression.
+    /// If that initializer expression is another local binding, find its initializer again.
+    ///
+    /// This process repeats as long as possible (but usually no more than once).
+    /// Type-check adjustments are not taken in account in this function.
+    ///
+    /// Examples:
+    /// ```
+    /// let abc = 1;
+    /// let def = abc + 2;
+    /// //        ^^^^^^^ output
+    /// let def = def;
+    /// dbg!(def);
+    /// //   ^^^ input
+    /// ```
+    pub fn expr_or_init<'a>(&self, mut expr: &'a hir::Expr<'tcx>) -> &'a hir::Expr<'tcx> {
```

Comment:
Isn't it a bit dangerous to change the other method's name and reusing it for this one?

Reply:
Yes, a bit, but the function was added [very recently](https://github.com/rust-lang/rust/pull/115257), has only one user (the utf8 lints) and it is now better matching the similar function from clippy: [`clippy_utils::expr_or_init`](https://doc.rust-lang.org/nightly/nightly-rustc/clippy_utils/fn.expr_or_init.html).